addons/sale/tests/test_sale_to_invoice.py

```diff
@@ -210,6 +210,49 @@ def test_downpayment_percentage_tax_icl(self):
         invoice.action_post()
         self.assertEqual(downpayment_line.price_unit, self.sale_order.amount_total/2, 'downpayment should have the correct amount')
 
+    def test_downpayment_invoice_and_partial_credit_note(self):
+        """This test check that the downpayment line amount on the sale order remains consistent"""
+        self.sale_order.action_confirm()
+
+        # Create an invoice for a Down payment of 100
+        product_id = self.env['ir.config_parameter'].sudo().get_param('sale.default_deposit_product_id')
+        product_id = self.env['product.product'].browse(int(product_id)).exists()
```

Comment:
Why the `exists()`? I would expect the test to fail if the id doesn't exist instead of potentially failing silently

Reply:
Bad copy pasta from line 191.
Noted to check for it